pkg/config/config.go

```diff
@@ -27,7 +30,40 @@ func NewBootloader(vmlinuzPath, kernelCmdLine, initrdPath string) *Bootloader {
 	}
 }
 
+func isCompressed(filename string) (bool, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	buf := make([]byte, 2048)
+	_, err = file.Read(buf)
+	if err != nil {
+		return false, err
+	}
+	kind, err := filetype.Match(buf)
+	if err != nil {
+		return false, err
+	}
+	// uncompressed ARM64 kernels are matched as a MS executable, which is
+	// also an archive, so we need to special case it
+	if kind == matchers.TypeExe {
+		return false, nil
+	}
+
+	return filetype.IsArchive(buf), nil
```

Comment:
Based on the comment this `filetype.IsArchive()` is not strictly necessary, right? we can directly `return true, nil`

Reply:
Right, some left-over from the initial implementation. Could be useful if some day the file detection code no longer detects the kernel as `matchers.TypeExe`. Let me update this.